sources/declarations.shen

```diff
@@ -185,12 +185,25 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
   [F | Y] X -> (append [F | Y] [X])
   F X -> [F X])
 
-(set *symbol-table*
-      [[datatype-error | (/. X (datatype-error X))]
-       [tuple | (/. X (tuple X))]
-       [pvar | (/. X (pvar X))]
-       |
-       (mapcan (/. X (symbol-table-entry X)) (external (intern "shen")))])
+(if (not (bound? *symbol-table*))
```

Comment:
This `if` here is so that implementations can pre-declare their own dictionary, which sometimes can be done more efficiently (for example, in Scheme I can make a hash table that works only with symbols and keys and is more efficient than a generic one).